python/src/space/core/storage.py

```diff
@@ -187,6 +187,40 @@ def transaction(self) -> Transaction:
     """Start a transaction."""
     return Transaction(self)
 
+  def lookup_tag(self,tag:str) -> meta.SnapshotReference:
+    if tag in self._metadata.tags:
+      return self._metadata.tags[tag]
+    raise errors.TagNotFoundError(f"Tag {tag} is not found")
+
+  def add_tag(self, tag:str, snapshot_id: Optional[int] = None) -> None:
+    """Add tag to a snapshot"""
+    if snapshot_id is None:
+      snapshot_id = self._metadata.current_snapshot_id
+    if snapshot_id not in self._metadata.snapshots:
+        raise errors.SnapshotNotFoundError(f"Snapshot {snapshot_id} is not found")
+    if tag in self._metadata.tags:
+        raise errors.TagExistError(f"Tag {tag} already exist")
+    new_metadata_path = self.new_metadata_path()
```

Comment:
ditto, please add an empty line after if block, otherwise for Py it is hard to read. Please check other places as well.

Reply:
Done